src/Elastic.Apm/Api/IRequest.cs

```diff
@@ -0,0 +1,11 @@
+using Elastic.Apm.Model.Payload;
+
+namespace Elastic.Apm.Api
+{
+	public interface IRequest
+	{
+		string HttpVersion { get; set; }
+		string Method { get; set; }
+		object Body { get; set; }
```

Comment:
Hi, I'm also not sure about that, i changed to object according to schema https://github.com/elastic/apm-server/blob/master/docs/spec/request.json#L9 .
As for me, having that as an object is not a good idea, because we can have issues with serialization as you already mentioned, and the user can have possible issues with index mappings. So for me i will always use string or dictionary<string,string>. 
But I`m not sure that we should limit library users in some strict types however apm server supports to have it as an object.

Reply:
Yeah, the problem is that `object` in the json schema is a json object and that's different from `object` in C#... we cannot map any C# `object` to a json object... or..well, we can, but we can't guarantee correctness.

Let's do the following: let's keep it `object` and if we see that this does not work we can simply change it before we GA the agent.

Reply:
Cool, agree